src/pudl/transform/eia.py

```diff
@@ -547,11 +540,30 @@ def harvesting(  # noqa: C901
     return entity_df, annual_df, col_dfs
 
 
-def _boiler_generator_assn(  # noqa: C901
-    eia_transformed_dfs: dict[str, pd.DataFrame],
-    eia923_years: list[int] | None = None,
-    eia860_years: list[int] | None = None,
-    debug: bool = False,
+@asset(
+    ins={
+        table_name: AssetIn()
+        for table_name in [
+            "clean_boiler_generator_assn_eia860",
+            "clean_generation_eia923",
+            "clean_generators_eia860",
+            "clean_boiler_fuel_eia923",
+        ]
+    },
+    config_schema={
+        "debug": Field(
+            bool,
+            default_value=False,
+            description=(
+                "If True, debugging columns will be added to boiler_generator_assn."
+            ),
+        ),
+    },
+    required_resource_keys={"dataset_settings"},
+    io_manager_key="pudl_sqlite_io_manager",
+)
+def boiler_generator_assn_eia860(  # noqa: C901
```

Comment:
I got rid of the wrapper function and lightly modified the internals of the BGA function to work as an asset on its own.